src/backend/catalog/pg_constraint.c

```diff
@@ -172,8 +177,17 @@ CreateConstraintEntry(const char *constraintName,
 		values[i] = (Datum) NULL;
 	}
 
-	conOid = GetNewOidWithIndex(conDesc, ConstraintOidIndexId,
-								Anum_pg_constraint_oid);
+	if (sql_dialect == SQL_DIALECT_TSQL && isTempNamespace(constraintNamespace))
```

Comment:
let us create a function in query env and use it here
if (useTempOidBuffer(constraintNamespace))

all condition can go inside useTempOidBuffer

Reply:
done